modules/javafx.controls/src/main/java/javafx/scene/control/TreeTableView.java

```diff
@@ -579,7 +579,7 @@ public static int getNodeLevel(TreeItem<?> node) {
         @Override public Boolean call(TreeTableView table) {
             try {
                 TreeItem rootItem = table.getRoot();
-                if (rootItem == null) return false;
+                if (rootItem == null | rootItem.getChildren().isEmpty()) return false;
```

Comment:
That should be `||` (boolean OR). In addition to being less clear, this will get an NPE if `rootItem` is ever null, since the bit-wise `|` operator doesn't short-curcuit the rest of the `if` test if the first term is true.

Given that this would have caused a regression, and that we don't have a test that would catch it, can you add a test that sets the root to null and calls sort? It will fail with this version of the fix, but would pass without the fix and should pass once you update your fix.

Reply:
Sorry about that, It was a typo. But it did reveal the missing test. 
Corrected the operator to `||` and added a test as you suggested which fails only with the commit 1 of this PR.